Lib/test/test_subprocess.py

```diff
@@ -1661,6 +1662,78 @@ def test_start_new_session(self):
             child_pgid = int(output)
             self.assertNotEqual(parent_pgid, child_pgid)
 
+    def test_user(self):
+        # For code coverage of the user parameter.  We don't care if we get an
+        # EPERM error from it depending on the test execution environment, that
+        # still indicates that it was called.
+
+        uid = os.getuid()
+        for user in [65534 if uid != 65534 else 65533,
+                     "nobody" if sys.platform != 'darwin' else "unknown"]:
+            try:
+                output = subprocess.check_output(
+                        [sys.executable, "-c",
+                        "import os; print(os.getuid())"],
+                        user=user)
+            except OSError as e:
+                if e.errno != errno.EPERM:
+                    raise
+            else:
+                parent_user = os.getuid()
+                child_user = int(output)
+                self.assertNotEqual(parent_user, child_user)
```

Comment:
Why do you check inequality with the uid of the parent instead of equality with the uid you really wanted to set? Same with the other test.

Reply:
Artifact of refactoring, the tests worked a little different originally. I will fix this.